File: https-quera.org-problemset-17675/Solution.java
import java.util.HashSet;
import java.util.Scanner;
import java.util.Set;

public class Solution {

    public static void main(String[] args) {
        app();
    }

    public static void app() {
        Scanner scanner = new Scanner(System.in);
        int num = scanner.nextInt();
        Set<Integer> fibs = new HashSet<>();
        for (int fib : fib(num)) {
            fibs.add(fib);
        }

        for (int i = 1; i <= num; i++) {
            if (fibs.contains(i)) {
                System.out.print("+");
            } else {
                System.out.print("-");
            }
        }
    }

    public static Iterable<Integer> fib(int n) {
        return () -> new java.util.Iterator<>() {
            private int a = 1, b = 1, count = 0;

            @Override
            public boolean hasNext() {
                return count < n;
            }

            @Override
            public Integer next() {
                int current = a;
                int next = a + b;
                a = b;
                b = next;
                count++;
                return current;
            }
        };
    }
}

File: https-quera.org-problemset-17675/Solution.py
def app():
    num = int(input())
    fibs = set(fib(num))
    for _ in range(1, num + 1):
        if _ in fibs:
            print("+", end="")
        else:
            print("-", end="")


def fib(n):
    a, b = 1, 1

    for i in range(n):
        yield a
        a, b = b, a + b


if __name__ == "__main__":
    app()